File: simple_talker.py
import rclpy
from rclpy.node import Node
from std_msgs.msg import String


class SimpleTalker(Node):
    def __init__(self) -> None:
        super().__init__("simple_talker")
        self.publisher_ = self.create_publisher(String, "chatter", 10)
        self.timer = self.create_timer(1.0, self.timer_callback)
        self.count = 0

    def timer_callback(self) -> None:
        msg = String()
        msg.data = f"Hello ROS 2! count={self.count}"
        self.publisher_.publish(msg)
        self.get_logger().info(f'Publish: "{msg.data}"')
        self.count += 1


def main(args=None) -> None:
    rclpy.init(args=args)
    node = SimpleTalker()
    try:
        rclpy.spin(node)
    except KeyboardInterrupt:
        pass
    finally:
        node.destroy_node()
        rclpy.shutdown()


if __name__ == "__main__":
    main()

File: simple_talker.cpp
#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/string.hpp"

class SimpleTalker : public rclcpp::Node
{
public:
  SimpleTalker() : Node("simple_talker"), count_(0)
  {
    publisher_ = this->create_publisher<std_msgs::msg::String>("chatter", 10);

    timer_ = this->create_wall_timer(
      std::chrono::seconds(1),
      std::bind(&SimpleTalker::timer_callback, this));
  }

private:
  void timer_callback()
  {
    auto msg = std_msgs::msg::String();
    msg.data = "Hello ROS2! count=" + std::to_string(count_);

    publisher_->publish(msg);

    RCLCPP_INFO(this->get_logger(), "Publish: '%s'", msg.data.c_str());

    count_++;
  }

  rclcpp::Publisher<std_msgs::msg::String>::SharedPtr publisher_;
  rclcpp::TimerBase::SharedPtr timer_;
  int count_;
};

int main(int argc, char * argv[])
{
  rclcpp::init(argc, argv);
  rclcpp::spin(std::make_shared<SimpleTalker>());
  rclcpp::shutdown();
  return 0;
}